daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/BinaryNumberTraits.scala

```diff
@@ -107,12 +106,16 @@ trait PrefixedLengthParserMixin {
    * returning, indicating that the returned value is meaningless.
    */
   def getPrefixedLengthInBits(state: PState): Long = {
-    Assert.invariant(lengthUnits != LengthUnits.Characters)
     val lenInUnits = getPrefixedLengthInUnits(state)
-    if (lengthUnits == LengthUnits.Bytes) {
-      lenInUnits * 8
-    } else {
-      lenInUnits
+    lengthUnits match {
```

Comment:
Is there a jira ticket for this - inability to support prefixLength in characters?  Because this fixes that, though not for variable-width encodings.